### PR TITLE
Build system

### PR DESCRIPTION
**Pull request by hildjj**
This is a rework of #107.  I simplified a few things, removed a few things that we're not using yet, and made sure that the web online, benchmark, and tests are using the new builds correctly.  Web tests still use mocha for the moment, but getting Jest to work in the browser isn't trivial, and `jest-lite` didn't work out of the box for me.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ module.exports = {
     "benchmark/vendor/",
     "browser/",
     "node_modules/",
-    "*.min.js"
+    "*.min.js",
+    "build",
+    "rollup.config.js" // in .eslintrc-modules.js
```
Comment (hildjj) [inline .eslintrc.js:15]: If we start using more modules, we can add '*.mjs' here, and use the second call to eslint to get those also.

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -11,33 +10,18 @@ on:
...
-      - name: Install pnpm
-        uses: pnpm/action-setup@v1.2.1
-        with:
-          version: 5.18.9
-          run_install: true
-      - name: Lint ${{ matrix.node-version }}
-        run: npm run lint
       - name: Test ${{ matrix.node-version }}
-        run: npm run test:bare
+        run: npm install && npm run build
```
Comment (hildjj) [inline .github/workflows/node.js.yml:27]: npm build runs lint now.

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+  "compilerOptions": {
+    "target": "es5",                                /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
```
Comment (hildjj) [inline tsconfig.json:3]: Changing this from es6 to es5 is what makes this a babel replacement step.  It makes the es6 output nearly useless however, so I removed that for the moment.

Comment (StoneCypher) [reply tsconfig.json:3]: oh, i forgot about this, sorry

it'd be best to emit them from separate paths through compilation.  es5 and es6 end up substantially different in size and speed due to the shims and shams involved, and the es6 downstreams can do a better job of tree shaking when unshimmed

**Comment by Mingun** [comment]
As I already said in #107, I do not like the way how it implemented. Splitting that to small, isolated steps will be better.

Objections were @StoneCypher based on the fact that this is impossible (or very, very difficult).

Looking at the final list of modified files in this PR, it seems highly exaggerated to me.

There are, at least, many fixup commit, many of them just undoing changes from the previous commit(s). For example, commit, that [adds](https://github.com/peggyjs/peggy/pull/142/commits/9979d55c653083b7e6fbd38b79483f324dc135ee) `coverage` folder under the version control (why?) and commit, that [removes](https://github.com/peggyjs/peggy/pull/142/commits/b25b9df13984b4606b39f74dda7070d48cc0c008) it.

[Deletion](https://github.com/peggyjs/peggy/pull/142/commits/9d2a92b62d85e8577f8f13ab13a70880a1cd2a87) of the node 10.x and [addition](https://github.com/peggyjs/peggy/pull/142/commits/6007b8fd680ef7edb6638dca7b03aa0b8552e799) it back in the next commit. The addition commit itself does too many work:
- replaces `rf -rf` with `rimraf`
- fixes running some npm tasks on Windows (by direct invoking `node` instead of shebang magic)
- changes npm scripts:
  - removes pretest step with parser regeneration
  - renames some steps
- removes `browserify` (and, transitively, `babelify`)
- adds typescript compiler + jest + rollup
  - even that work is done not completely, because the [next commit](https://github.com/peggyjs/peggy/pull/142/commits/a26c3d5dadd91c9060bc281e83d177dfcb767445) does some fixes in it
  - another [fixup](https://github.com/peggyjs/peggy/pull/142/commits/7cccf9a7fb5625c40c1876d8717e19f77abc8a85) for `jest` mixed up with the version autoupdating. That commit also deletes files, that was added in the previous jest commit

Some [strange](https://github.com/peggyjs/peggy/pull/142/commits/a4c4d9acfd6560cb2ef3117a09a960cf5062ae5b) [commits](https://github.com/peggyjs/peggy/pull/142/commits/b5585835dc327b9d2cd375aea9c23b843cb06583) returning stuff, removed by the couple commits ago.

[Yet another](https://github.com/peggyjs/peggy/pull/142/commits/5dfe18f8e6ee66d73fd456b34152f565e108751f) round of throwing [back](https://github.com/peggyjs/peggy/pull/142/commits/6d79b2fc7af710899724230c014774cc0a47fcc7) and [forth](https://github.com/peggyjs/peggy/pull/142/commits/3fb5e277cd1471ddf08f170147471dc418c7e0b4)

[Clearly an experimental commit](https://github.com/peggyjs/peggy/pull/142/commits/244c3c4a266070e646269358481092d0b7760678) that shouldn't be in history.

[Fixup](https://github.com/peggyjs/peggy/pull/142/commits/7a9e97f3bc6f390b7396d6069c77372f03a1d1d5) [commits](https://github.com/peggyjs/peggy/pull/142/commits/8a67d82130931c62e423cd1d6765ed4fdbfebe15), [yet](https://github.com/peggyjs/peggy/pull/142/commits/c32aac69258065de891be35435b0ebb04d26736d) [another](https://github.com/peggyjs/peggy/pull/142/commits/fde3ed812e5bc2e8df13bd81aa2fd17cb4d06cb9) [fixups](https://github.com/peggyjs/peggy/pull/142/commits/49552c1754eefff5a0219527ca2133c71a9e9cb8)

-----

I believe that upgrade can be made in more understandible way. If not decoupled into small pieces, than just in one commit that removes everything and another that adds the new infrastructure.

**Comment by hildjj** [comment]
I think the best I can do at this point is squash-merge 107 then drop this on top.  I think that gets us to a better place than we were, and we can move forward from there.

**Comment by hildjj** [comment]
I'm going to follow the above plan in an hour or so unless anyone complains.

**Comment by hildjj** [comment]
... except I can't squash merge 107 since the original repo is private now.  I'm tempted to just squash this whole thing, commit it as me, and make sure @StoneCypher still gets credit in the comments and release notes.

**Comment by hildjj** [comment]
I think I can add a Co-authored-by in the commit.

**Comment by hildjj** [comment]
OK, I think I got all of the commits squashed, and attributed to the right people (probably off by a line or two while i dealt with conflicts).  I'm waiting for the tests to pass, then will take another look through to see if everything looks right.

**Comment by hildjj** [comment]
Fixes #100 
Fixes #96 
Fixes #94 
Fixes #83
Fixes #75 
Fixes #90 
Fixes #77

**Comment by hildjj** [comment]
Everything looks good, I'm pulling the trigger.

File: .eslintrc.js (hunk 1 above)
Comment (StoneCypher) [reply .eslintrc.js:15]: no, mjs doesn't exist

**Comment by Mingun** [comment]
OK with that. If I had more time, I would have time to study the tools and would have tried to do what I described, but did not have time 🤷

**Pull request merged**